pkg/sources/git/git.go

```diff
@@ -615,10 +620,16 @@ func (s *Git) gitChunk(ctx context.Context, diff gitparse.Diff, fileName, email,
 
 // ScanStaged chunks staged changes.
 func (s *Git) ScanStaged(ctx context.Context, repo *git.Repository, path string, scanOptions *ScanOptions, reporter sources.ChunkReporter) error {
-	// Get the URL metadata for reporting (may be empty).
-	urlMetadata := getSafeRemoteURL(repo, "origin")
+	// Get the remote URL for reporting (may be empty)
+	remoteURL := getSafeRemoteURL(repo, "origin")
+	var repoCtx context.Context
+	if remoteURL != "" {
+		repoCtx = context.WithValue(ctx, "repo", remoteURL)
+	} else {
+		repoCtx = ctx
+	}
 
-	commitChan, err := gitparse.NewParser().Staged(ctx, path)
+	commitChan, err := gitparse.NewParser().Staged(repoCtx, path)
```

Comment:
Idk if there's value in including the remote url when scanning staged changes. Presumably this will only be called in the context of a single repository, so the culprit should be obvious.